src/http/Request.ts

```diff
@@ -1,27 +1,33 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License.
 
-import { HttpMethod, HttpRequest } from '@azure/functions';
+import { HttpMethod, HttpRequest, HttpRequestHeaders, HttpRequestParams, HttpRequestQuery } from '@azure/functions';
+import { AzureFunctionsRpcMessages as rpc } from '../../azure-functions-language-worker-protobuf/src/rpc';
+import { fromTypedData } from '../converters/RpcConverters';
+import { fromNullableMapping, fromRpcHttpBody } from '../converters/RpcHttpConverters';
 
-export class RequestProperties implements HttpRequest {
-    method: HttpMethod | null = null;
-    url = '';
-    originalUrl = '';
-    headers: { [key: string]: string } = {};
-    query: { [key: string]: string } = {};
-    params: { [key: string]: string } = {};
-    body?: any;
-    rawBody?: any;
-    [key: string]: any;
-}
+export class Request implements HttpRequest {
+    public method: HttpMethod | null;
+    public url: string;
+    public originalUrl: string;
+    public headers: HttpRequestHeaders;
+    public query: HttpRequestQuery;
+    public params: HttpRequestParams;
+    public body?: any;
+    public rawBody?: any;
 
-export class Request extends RequestProperties {
-    constructor(httpInput: RequestProperties) {
-        super();
-        Object.assign(this, httpInput);
+    public constructor(rpcHttp: rpc.IRpcHttp) {
+        this.method = <HttpMethod>rpcHttp.method;
+        this.url = <string>rpcHttp.url;
+        this.originalUrl = <string>rpcHttp.url;

```

Comment:
I know this was also in the original version, but is there a difference between `this.url` and `this.originalUrl`?

Reply:
Nope. There's quite a bit of duplicate stuff, but we can't clean it up because it might break people